rita_exit/src/network_endpoints/mod.rs

```diff
@@ -191,119 +188,49 @@ pub async fn get_exit_timestamp_http(_req: HttpRequest) -> HttpResponse {
-    let exit_settings = get_rita_exit();
-    let our_secretkey: WgKey = exit_settings.network.wg_private_key.unwrap();
-    let our_secretkey = our_secretkey.into();
-
-    let their_nacl_pubkey = request.pubkey.into();
-
-    let contact = Web3::new(&get_web3_server(), CLIENT_STATUS_TIMEOUT);
+/// This exit may not be included in the list returned by the smart contract! If this is the case it must
+/// first be added to the root of trust server, and clients will know to choose a different exit.
+pub async fn get_exit_list() -> HttpResponse {
     let rita_exit = get_rita_exit();
-    let our_id = rita_exit.get_identity().unwrap();
-    let our_addr = rita_exit
-        .payment
-        .eth_private_key
-        .expect("Why do we not have a private key?")
-        .to_address();
     let contract_addr = rita_exit.exit_network.registered_users_contract_addr;
 
-    let ret: ExitList = ExitList {
-        exit_list: match get_exits_list(&contact, our_addr, contract_addr).await {
-            Ok(a) => {
-                let exit_regions = rita_exit.exit_network.allowed_countries;
-
-                // only one payment type can be accepted for now, but this structure allows for
-                // multiple payment types in the future
-                let mut accepted_payments = HashSet::new();
-                accepted_payments.insert(exit_settings.payment.system_chain);
-
-                if exit_regions.is_empty() || accepted_payments.is_empty() {
-                    error!("Exit list not configured correctly. Please set up exit regions and accepted payment types in config");
-                    return HttpResponse::InternalServerError().finish();
-                }
-                let mut ret = vec![];
-                for exit in a {
-                    // Remove Exits that dont have proper regions defined
-                    let mut exit_allowed_regions = exit.allowed_regions.clone();
-                    if exit_allowed_regions.remove(&Regions::UnkownRegion) {
-                        warn!("Found an uknown region in exit! {:?}", exit);
-                    }
-
-                    if exit_allowed_regions.is_empty() || exit.payment_types.is_empty() {
-                        error!(
-                            "Invalid configured exit, no allowed regions or payments setup! {:?}",
-                            exit
-                        );
-                        continue;
-                    }
-                    if !exit_allowed_regions.is_disjoint(&exit_regions)
-                        && !exit.payment_types.is_disjoint(&accepted_payments)
-                    {
-                        ret.push(exit_identity_to_id(exit))
-                    }
-                }
-                ret.push(our_id); // add ourselves to the list
-                ret
+    // we are receiving a SignedExitServerList from the root server.
+    let signed_list: SignedExitServerList =
+        match get_exit_list_from_root(contract_addr).await {
```

Comment:
this should have a cache just like the root of trust server does. That way if the exit experiences a momentary blip in connectivity, or say the exit root of trust server goes down for a few seconds, or even hours, while maint is happening it won't disrupt normal operation.